tutorials/run-spring-petclinic-on-app-engine-flex-cloudsql.md

```diff
@@ -0,0 +1,131 @@
+---
+title: Run Spring Pet Clinic with CloudSQL on Google App Engine Flexible Environment
+description: Learn how to deploy Spring Boot Pet Clinic application to Google App Engine flexible environment and use CloudSQL.
+author: jishaa
+tags: App Engine Flex, CloudSQL, Spring Boot
+date_published: 2017-05-12
+---
+
+This tutorial will walk you through getting Spring Pet Clinic application up and running on App Engine Flex with CloudSQL.
+
+### Spring Boot and Spring Pet Clinic
+[Spring Boot](https://projects.spring.io/spring-boot/) provides Java developers a quick annotation driven way to deploy services using minimal code.
+[Spring Pet Clinic](https://github.com/spring-projects/spring-petclinic) is a popular Spring application that will quickly demonstrate the power of SpringBoot.
+
+### App Engine Flex environment for Java
+[App Engine Flex](https://cloud.google.com/appengine/docs/flexible/java/) provide the ability to run your Java applications in Docker containers on GCE machines, with the ability
+to have them deployed, monitored and auto-scaled.
+We will be using the [Java 8 Runtime environment](https://cloud.google.com/appengine/docs/flexible/java/dev-java-only) for this tutorial.
+
+### CloudSQL
+[CloudSQL for MySQL](https://cloud.google.com/sql/docs/mysql/) is a fully managed MySQL database service on GCP.
+
+### Prerequisites
+1. Create a project in the [Google Cloud Platform Console](https://console.cloud.google.com/).
+1. Enable billing for your project.
+1. Install the [Google Cloud SDK](https://cloud.google.com/sdk/).
+
+### Prepare
+1. Initialize, create an app engine application and authorize `gcloud` for using GCP APIs in your local environment :
+```
+   gcloud init
+   gcloud app create
+   gcloud auth application-default login
+```
+1. Clone and test the Pet Clinic application locally :
+```
+    git clone https://github.com/spring-projects/spring-petclinic.git
+    cd spring-petclinic
+    ./mvnw spring-boot:run
+```
+
+  Access http://localhost:8080 via your web browser to see the application homepage.
+
+  By default, the application uses an [hsqldb](http://hsqldb.org/), an in-memory database.
+  We will now switch from `hsqldb` to using CloudSQL as your database.
+
+### Using CloudSQL as your database
+
+1. Enable the [Cloud SQL API](https://console.cloud.google.com/flows/enableapi?apiid=sqladmin).
+
+1. Create a 2nd Generation CloudSQL (MySQL) instance and set the root user password following the instructions [here](https://cloud.google.com/sql/docs/mysql/create-instance#create-2nd-gen).
```

Comment:
s/2nd/Second/

also, s/following the instructions [here]/by following [these instructions]/